rules-engine/src/rules_engine/engine.py

```diff
@@ -329,6 +329,53 @@ def _initialize_billing_periods(self, billing_periods: List[BillingPeriod]) -> N
         self.bills_summer = []
         self.bills_shoulder = []
 
+
+        # #####
+        # Logic Change:
+        # 'Inclusion' in calculations is now determined by two variables:
+        # billing_period.inclusion_calculated: bool 
+        # billing_period.inclusion_override: bool (False by default)
+        # 
+        # Our logic around the AnalysisType can now disallow
+        # a billing_period from being included or overridden
+        # by marking it as NOT_ALLOWED_IN_CALCULATIONS
+        # 
+        # Options for billing_period.analysis_type:
+        # ALLOWED_HEATING_USAGE = 1 # winter months - allowed in heating usage calculations
+        # ALLOWED_NON_HEATING_USAGE = -1 # summer months - allowed in non-heating usage calculations
+        # NOT_ALLOWED_IN_CALCULATIONS = 0 # shoulder months that fall outside reasonable bounds 
+        # 
+        # Use HDDs to determine if shoulder months
```

Comment:
I proposed this approach during discussion, but let's defer implementing it until after the MVP release. A change like this could make it difficult to verify the Rules Engine vs. the original spreadsheet.